### PR TITLE
zh-CN: update webassembly/concepts

### LINKED ISSUE #19467: Some links are currently 404
### MDN URL

https://developer.mozilla.org/zh-CN/docs/WebAssembly/Concepts

### 与此议题相关的小节或标题？

从 C/C++ 移植

### 哪些信息是错误、无用或不完善的？

image

### 你期望看到什么？

Remove the link for 404       

### 你有什么支持的链接、参考或引用？

_No response_

### 你还想补充什么内容？

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `zh-cn/webassembly/concepts`
* MDN URL: https://developer.mozilla.org/zh-CN/docs/WebAssembly/Concepts
* GitHub URL: https://github.com/mdn/translated-content/blob/main/files/zh-cn/webassembly/concepts/index.md
* Last commit: https://github.com/mdn/translated-content/commit/15dbba38ed33e3d697c2ee747f14b06d98cf8c7f
* Document last modified: 2023-12-26T13:21:46.000Z

</details>

### PR DESCRIPTION
**Pull request by PassionPenguin**
Related update in original content:

https://github.com/mdn/content/commit/b9a41e258e656b57f41e8aaf88f93e51bb27974d

Fix #19467

### CODE REVIEW

File: files/zh-cn/webassembly/concepts/index.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 title: WebAssembly 概念
 slug: WebAssembly/Concepts
+l10n:
+  sourceCommit: b9a41e258e656b57f41e8aaf88f93e51bb27974d
```
Comment (yin1999) [inline files/zh-cn/webassembly/concepts/index.md:5]: Do not add this, I believe the current document is not in sync with the full content of the corresponding English commit.

Comment (PassionPenguin) [reply files/zh-cn/webassembly/concepts/index.md:5]: oops, i believe so.

**Comment by PassionPenguin** [comment]
PTAL @yin1999

File: files/zh-cn/webassembly/concepts/index.md [hunk 2]
```diff
@@ -40,88 +42,84 @@ WebAssembly 是一种运行在现代网络浏览器中的新型代码，并且
...
 你可以在 [emscripten.org](http://emscripten.org) 找到关于 Emscripten 的完整文档以及在[将 C/C++ 编译为 WebAssembly](/zh-CN/docs/WebAssembly/C_to_wasm) 找到一个实现工具链并交叉编译你自己的 C/C++ 应用程序为 wasm 的指南。
 
 ### 直接编写 WebAssembly 代码
 
 你想构建自己的编译器，或者你自己的工具，或者创建一个能够在运行时生成 WebAssembly 代码的 JavaScript 库？
 
-就像真实的汇编语言一样，WebAssembly 的二进制格式也有文本表示——两者之间 1:1 对应。你可以手工书写或者生成这种格式然后使用这些工具（[WebAssemby text-to-binary tools](http://webassembly.org/getting-started/advanced-tools/)）中的任何一个把它转换为二进制格式。
+就像真实的汇编语言一样，WebAssembly 的二进制格式也有文本表示——两者之间 1:1 对应。你可以手工书写或者生成这种格式然后使用这些工具（[WebAssembly 文本转化为二进制工具](http://webassembly.org/getting-started/advanced-tools/)）中的任何一个把它转换为二进制格式。
+
+有关如何完成此操作的简单指南，请参阅我们的[转换 WebAssembly 文本格式为 Wasm](/zh-CN/docs/WebAssembly/Text_format_to_wasm)指南。
```
Comment (jasonren0403) [inline files/zh-cn/webassembly/concepts/index.md:114]:
```suggestion
有关如何完成此操作的简单指南，请参阅我们的[转换 WebAssembly 文本格式为 Wasm](/zh-CN/docs/WebAssembly/Text_format_to_wasm) 指南。
```

File: files/zh-cn/webassembly/concepts/index.md [hunk 3]
```diff
@@ -40,88 +42,84 @@ WebAssembly 是一种运行在现代网络浏览器中的新型代码，并且
...
+
+### 编写以 WebAssembly 为目标的 Rust 代码
+
+在 Rust WebAssembly 工作组（Rust WebAssembly Working Group）的不懈努力下，编写 Rust 代码并编译到 WebAssembly 也成为可能。你可以在我们的[从 Rust 编译到 WebAssembly](/zh-CN/docs/WebAssembly/Rust_to_Wasm)一文中开始安装必要的工具链，将 Rust 示例程序编译到 WebAssembly npm 包中，并在示例 Web 应用程序中使用。
```
Comment (jasonren0403) [inline files/zh-cn/webassembly/concepts/index.md:118]:
```suggestion
在 Rust WebAssembly 工作组（Rust WebAssembly Working Group）的不懈努力下，编写 Rust 代码并编译到 WebAssembly 也成为可能。你可以在我们的[从 Rust 编译到 WebAssembly](/zh-CN/docs/WebAssembly/Rust_to_Wasm) 一文中开始安装必要的工具链，将 Rust 示例程序编译到 WebAssembly npm 包中，并在示例 Web 应用程序中使用。
```

**Review by jasonren0403 — COMMENTED** [review]
suggestions

File: files/zh-cn/webassembly/concepts/index.md [hunk 4]
```diff
@@ -11,9 +13,9 @@ slug: WebAssembly/Concepts
 
 WebAssembly 是一种运行在现代网络浏览器中的新型代码，并且提供新的性能特性和效果。它设计的目的不是为了手写代码而是为诸如 C、C++ 和 Rust 等低级源语言提供一个高效的编译目标。
 
-对于网络平台而言，这具有巨大的意义——这为客户端 app 提供了一种在网络平台以接近本地速度的方式运行多种语言编写的代码的方式；在这之前，客户端 app 是不可能做到的。
+对于网络平台而言，这具有巨大的意义——这为客户端 App 提供了一种在网络平台以接近本地速度的方式运行多种语言编写的代码的方式；在这之前，客户端 App 是不可能做到的。
```
[the remark this answers is not in the document]
Comment (PassionPenguin) [reply files/zh-cn/webassembly/concepts/index.md:16]:
```suggestion
对于 Web 平台而言，这具有巨大的意义——这为客户端 App 提供了一种在 Web 平台以接近本地速度的方式运行多种语言编写的代码的方式；在这之前，客户端 App 是不可能做到的。
```

**Review by jasonren0403 — APPROVED** [review]
LGTM 👍

**Pull request merged**